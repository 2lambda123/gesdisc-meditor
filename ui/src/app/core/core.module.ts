import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
