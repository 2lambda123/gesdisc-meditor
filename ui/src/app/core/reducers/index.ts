import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
