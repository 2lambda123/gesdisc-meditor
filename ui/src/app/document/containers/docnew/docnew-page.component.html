<div fxLayout="row" class="doc-new-container">
	<mat-card class="mat-elevation-z0" fxFlex="100">
		<med-doc-breadcrumbs [model]="model$ | async"></med-doc-breadcrumbs>

		<mat-card-title>
			<i class="icon-badge icon-badge-sm fa {{(model$ | async)?.icon?.name }}" [style.background-color]="(model$ | async)?.icon?.color"></i>
			{{(model$ | async)?.name}}
		</mat-card-title>

		<mat-card-subtitle>
			{{(model$ | async)?.description}}
		</mat-card-subtitle>
		
		<mat-card-content>
			<med-document-edit
				[document]="model$ | async"
				(liveData)="liveData($event)"
				(isValid)="isValid($event)">
			</med-document-edit>
			<button mat-raised-button 
        *ngIf="(userPrivileges$ | async).includes('edit')" 
        (click)="saveDocument()" 
        [disabled]="!isFormValid">Save
      </button>
		</mat-card-content>
	</mat-card>
</div>
