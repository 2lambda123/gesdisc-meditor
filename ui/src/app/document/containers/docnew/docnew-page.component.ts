import { Component, OnInit } from '@angular/core';

import * as fromApp from '../../../store';
import * as fromDocument from '../../store';
import { Model } from '../../../service/model/model';

import { Store, select } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';

@Component({
  selector: 'med-docnew-page',
  templateUrl: './docnew-page.component.html',
  styleUrls: ['./docnew-page.component.css']
})
export class DocNewPageComponent implements OnInit {

	modelName: string;
	titleProperty: string;
	model$: Observable<Model>;

  

	constructor(		
		private store: Store<fromApp.AppState>
	) {
	}

	ngOnInit() {	
		this.model$ = this.store.pipe(select(fromApp.getCurrentModel));		
		this.model$.subscribe(model => {
			this.modelName = model.name;
			this.titleProperty = model.titleProperty || 'title';
		});		
	}

	submitDocument(data: any) {
		data['x-meditor'] = {
			model: this.modelName
		};
    this.store.dispatch(new fromDocument.SubmitDocument(data));
	}	

}
