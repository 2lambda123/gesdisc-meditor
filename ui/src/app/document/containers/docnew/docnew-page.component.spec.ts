import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { DocnewComponent } from './docnew.component';

describe('DocnewComponent', () => {
  let component: DocnewComponent;
  let fixture: ComponentFixture<DocnewComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ DocnewComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DocnewComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
