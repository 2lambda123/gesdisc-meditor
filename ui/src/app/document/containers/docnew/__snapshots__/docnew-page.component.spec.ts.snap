// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document New Page should compile 1`] = `
<med-docnew-page
  __model$__selector={[Function Observable]}
  dirty="false"
  store={[Function Store]}
>
  <div
    class="doc-new-container"
    fxlayout="row"
  >
    <mat-card
      class="mat-elevation-z0"
      fxflex="100"
    >
      <med-doc-breadcrumbs />
      <mat-card-title>
        <i
          class="icon-badge icon-badge-sm fa "
        />
          
      </mat-card-title>
      <mat-card-subtitle>
          
      </mat-card-subtitle>
      <mat-card-content>
        <med-document-edit />
        
        <button
          disabled=""
          mat-raised-button=""
        >
          Save 
        </button>
      </mat-card-content>
    </mat-card>
  </div>
</med-docnew-page>
`;
